mod_reforged/hooks/ai/tactical/behaviors/ai_attack_throw_net.nut

```diff
@@ -4,6 +4,18 @@
 	q.onEvaluate = @(__original) function( _entity )
 	{
 		local ret = __original(_entity);
+		// It is a generator in vanilla, but we add the typeof check for maximum
+		// compatibility if another mod changed the original function to not be a generator.
+		if (typeof ret == "generator")
+		{
+			local thread = ret;
+			do
+			{
+				ret = resume thread;
+			}
+			while (ret == null);
+		}
+
 		if (this.m.Skill != null && ret != ::Const.AI.Behavior.Score.Zero && ::Math.rand(1, 100) < 75 && _entity.getSkills().hasSkill("perk.rf_kingfisher"))
```

Comment:
Its not very performant to let the onEvaluate run through completely only to discard the result at the end on a random roll. Instead that random roll and the other checks here could be done once after the very first `resume`.

Example
```
q.onEvaluate = @(__original) function( _entity )
{
	local ret = __original(_entity);
	// It is a generator in vanilla, but we add the typeof check for maximum
	// compatibility if another mod changed the original function to not be a generator.
	if (typeof ret == "generator")
	{
		local thread = ret;
		ret = resume thread;
		if (this.m.Skill != null && ret != ::Const.AI.Behavior.Score.Zero && ::Math.rand(1, 100) < 75 && _entity.getSkills().hasSkill("perk.rf_kingfisher"))
			ret = ::Const.AI.Behavior.Score.Zero;
		while (ret == null);
		{
			yield null;
			ret = resume thread;
		}
	}
	else
	{
		if (this.m.Skill != null && ret != ::Const.AI.Behavior.Score.Zero && ::Math.rand(1, 100) < 75 && _entity.getSkills().hasSkill("perk.rf_kingfisher"))
			ret = ::Const.AI.Behavior.Score.Zero;
	}
	
	return ret;
```